packages/beacon-node/src/chain/blocks/verifyBlock.ts

```diff
@@ -98,12 +104,16 @@ export async function verifyBlocksInEpoch(
             mergeBlockFound: null,
           } as SegmentExecStatus),
 
+      // data availability for the blobs
+      verifyBlocksDataAvailability(this, blocksInput, opts),
+
       // Run state transition only
       // TODO: Ensure it yields to allow flushing to workers and engine API
       verifyBlocksStateTransitionOnly(
         preState0,
         blocksInput,
-        dataAvailabilityStatuses,
+        // hack availability for state transition eval as availability is separately determined
+        blocks.map(() => DataAvailableStatus.available),
```

Comment:
should base on `block.slot` to determine `DataAvailableStatus` `preDeneb` vs `available`?

Reply:
internally in state transition it doesn't matter, but for correctness sake may be we can do it?